### PR TITLE
GH-12 - implement get brewery api

### PR DESCRIPTION
**Pull request by noebrito**
Implements a Get Recipe API. Added integration test. 

Ran locally and looks good.

```
~/workspace/openbrewerydb-java-api (GH-12) $ curl http://localhost:10000/breweries/1
{"id":1,"name":"testName","breweryType":"testBreweryType","city":"testCity","state":"testState","country":"testcountry","location":{"latitude":1.0,"longitude":1.0}}
```

### CODE REVIEW

File: src/main/java/org/openbrewerydb/AppServer.java [hunk 1]
```diff
@@ -20,6 +30,14 @@ public void run(final int port) {
     final Javalin app = Javalin.create()
         .start(port);
 
+    app.routes(() -> {
+      path("breweries", () -> {
+        path(":id", () -> {
+          get(this.breweryApi::getBrewery);
+        });
+      });
+    });
```
Comment (noebrito) [inline src/main/java/org/openbrewerydb/AppServer.java:39]: This is where routes will be defined.

File: src/main/java/org/openbrewerydb/api/BreweryApi.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package org.openbrewerydb.api;
+
+import io.javalin.http.Context;
+import org.openbrewerydb.models.Brewery;
+import org.openbrewerydb.service.BreweryService;
+
+/**
+ * Request handlers for Brewery API.
+ */
+public class BreweryApi {
```
Comment (noebrito) [inline src/main/java/org/openbrewerydb/api/BreweryApi.java:10]: Request handlers are defined in this class.

File: src/test/java/org/openbrewerydb/api/BreweryApiTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+@Testcontainers
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class BreweryApiTest implements WithAssertions {
+
+  @Container
+  public static PostgreSQLContainer<?> postgresDb = new PostgreSQLContainer<>(
+      DockerImageName.parse("postgis/postgis:12-2.5-alpine").asCompatibleSubstituteFor("postgres")
+  )
+      .withCommand("postgres -c max_connections=42")
```
Comment (chrisjm) [inline src/test/java/org/openbrewerydb/api/BreweryApiTest.java:35]: Life, universe, everything?

File: build.gradle [hunk 4]
```diff
@@ -31,13 +31,20 @@ dependencies {
     // logging
     implementation 'ch.qos.logback:logback-classic:1.2.3'
 
+    // json
+    implementation 'com.fasterxml.jackson.core:jackson-core:2.12.3'
```
Comment (chrisjm) [inline build.gradle:35]: The naming of things is definitely one of the more confusing things about Java for me. No way I would have guessed this library is for JSON. 😆

**Review by chrisjm — APPROVED** [review]
Looks good and mostly makes sense to a non-Java developer. I can generally follow the flow and makes sense to include the create and delete methods in this push due to testing. But in general, I have no idea so I'm taking your word for it. 😄

File: build.gradle (hunk 4 above)
Comment (noebrito) [reply build.gradle:35]: Haha. Yeah. Jackson and GSON are the two popular JSON serializers in Java. I use GSON at work so naturally I wanted to try the other one here...

File: src/test/java/org/openbrewerydb/api/BreweryApiTest.java (hunk 3 above)
Comment (noebrito) [reply src/test/java/org/openbrewerydb/api/BreweryApiTest.java:35]: Heh?

**Comment by noebrito** [comment]
> Looks good and mostly makes sense to a non-Java developer. I can generally follow the flow and makes sense to include the create and delete methods in this push due to testing. But in general, I have no idea so I'm taking your word for it. 😄 

Haha! Feel free to ask any questions at any time!

File: src/test/java/org/openbrewerydb/api/BreweryApiTest.java (hunk 3 above)
Comment (chrisjm) [reply src/test/java/org/openbrewerydb/api/BreweryApiTest.java:35]: Why is the `max_connections` equal to 42?

Comment (noebrito) [reply src/test/java/org/openbrewerydb/api/BreweryApiTest.java:35]: Honestly I don't remember. I copied it from an existing project. Let me look into it and get back to you.

**Pull request merged**